src/providers/UserStateProvider.tsx

```diff
@@ -15,10 +15,12 @@ import { Contract } from "ethcall"
 import ERC20_ABI from "../constants/abis/erc20.json"
 import { Erc20 } from "./../../types/ethers-contracts/Erc20.d"
 import { GaugeContext } from "./GaugeProvider"
+import { IS_DEVELOPMENT } from "../utils/environment"
 import { NETWORK_NATIVE_TOKENS } from "../constants/networks"
 import { TokensContext } from "./TokensProvider"
 import { Web3Provider } from "@ethersproject/providers"
 import { Zero } from "@ethersproject/constants"
+import { getTokenPrice } from "../utils/updateTokenPrices"
```

Comment:
I don't think this PR needs to modify the UserStateProvider at all, most of the logic should live in `updateTokenPrices`

Reply:
yeah, this was a testing to check the result!